dev_guide/templates.adoc

```diff
@@ -733,6 +733,62 @@ follows:
 }
 ====
 
+[[waiting-for-template-readiness]]
+=== Waiting for Template Readiness
+
+Template authors can optionally indicate that certain objects within a template
+should be waited for before a template instantiation via the Service Catalog /
+Template Service Broker / TemplateInstance API is considered complete.  This is
+an alpha feature and the API and implementation are subject to change.  Feedback
+is welcomed.
+
+To use this feature, a template author should mark one or more objects of kind
+`Build`, `BuildConfig`, `Deployment`, `DeploymentConfig`, `Job` or `StatefulSet`
+in a template with the annotation
+`"template.alpha.openshift.io/wait-for-ready": "true"`.  Template instantiation
+will not complete until all objects marked with the above annotation report
+ready.  Similarly, if any of the annotated objects report failed, or if the
+template fails to become ready within a fixed timeout of one hour, the template
+instantiation will fail.
```

Comment:
can you add a sample?  I worry people are not going to put the annotation in the right place otherwise.